app/models/conversation.rb

```diff
@@ -18,4 +18,25 @@ class Conversation < ApplicationRecord
   has_many :user_conversations
   has_many :users, through: :user_conversations
   has_many :messages, dependent: :destroy
+
+  def unread_messages_count(user)
+    unread_messages(user).count
+  end
+
+  def last_message
+    messages.order('created_at DESC').last
```

Comment:
Move this to a scope in the `Message` model.
Also, use `id` instead of `created_at` since there is an index for that attribute. And rewrite like: `.order(:id, :desc)`

Reply:
Is the last_message necessary? Because in the conversation list, you can always show the last message and the unread_count.
In that case, you can avoid having the `read` attribute, and store an `unread_count` in the conversation so you don't need to calculate the count every time the user wants to see the conversations.